### PR TITLE
Minor tweaks on Windows.

### PR DESCRIPTION
**Pull request by dkorolev**

### CODE REVIEW

File: Bricks/net/tcp/impl/posix.h [hunk 1]
```diff
@@ -296,7 +296,7 @@ class Connection : public SocketHandle {
   }
 
   inline Connection& BlockingWrite(const void* buffer, size_t write_length, bool more) {
-#ifdef CURRENT_APPLE
+#if defined(CURRENT_APPLE) || defined(CURRENT_WINDOWS)
     static_cast<void>(more);  // Supress the 'unused parameter' warning on Apple.
```
Comment (mzhurovich) [inline Bricks/net/tcp/impl/posix.h]: `s/on Apple//`?

File: Storage/transaction_result.h [hunk 2]
```diff
@@ -33,7 +33,12 @@ namespace storage {
 template <typename T>
 class TransactionResult : public OptionalResult<T> {
  public:
+#ifndef CURRENT_WINDOWS
   TransactionResult() = delete;
+#else
+  // MSVS needs this for `_Associated_state<>`, required by `std::promise<>`.
```
Comment (mzhurovich) [inline Storage/transaction_result.h]: :disappointed:

File: Bricks/net/tcp/impl/posix.h (hunk 1 above)
Comment (dkorolev) [inline Bricks/net/tcp/impl/posix.h]: :+1: 568dd8e

File: Storage/transaction_result.h (hunk 2 above)
Comment (dkorolev) [inline Storage/transaction_result.h]: Yes, but given we build & run on non-Windows every single time, I don't think this "sacrifice" is in any way big.

**Comment by mzhurovich** [comment]
@dkorolev :+1:

**Pull request merged**